adminapi/request.py

```diff
@@ -91,7 +91,14 @@ def calc_signature(private_key, timestamp, data=None):
     Returns the signature as base64 encoded unicode, ready for transport.
     """
     message = calc_message(timestamp, data)
-    sig = private_key.sign_ssh_data(message.encode())
+    try:
+        sig = private_key.sign_ssh_data(message.encode())
+    except SSHException as e:
+        # The key is not necessary could be used for authentication
```

Comment:
That sentence does not make sense. What do you want to say?
"The key provided cannot be used for authentication"?

Reply:
As PGP keys, they can be used for signature, encryption, or authentication. That's why the fact that the key is found in the agent doesn't mean, it must be able to sign something.